response-dto/src/main/java/org/ehrbase/openehr/sdk/response/dto/MetaData.java

```diff
@@ -17,11 +17,80 @@
  */
 package org.ehrbase.openehr.sdk.response.dto;
 
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+import org.ehrbase.openehr.sdk.response.dto.ehrscape.QueryResultDto;
 
+/**
+ * Represents a query result <code>meta</code> object.
+ *
+ * @see <a href="https://specifications.openehr.org/releases/ITS-REST/latest/query.html#tag/RESULT_SET_schema">Query API - RESULT_SET_schema</a>
+ */
 public class MetaData {
 
+    /**
+     * {@link MetaData} type for an AQL result.
+     */
+    public static final String RESULTSET = "RESULTSET";
+
+    /**
+     * Allows to store and retrieve {@link MetaData} <code>additional property</code> in a type safe manner.
+     *
+     * @see MetaData#setAdditionalProperty(String, Object)
+     * @param <T> of the additional property.
+     */
+    public interface AdditionalProperty<T> extends Function<Object, T> {
+
+        /**
+         * name of the JSON property field.
+         */
+        String getName();
+
+        @FunctionalInterface
+        interface LongProperty extends AdditionalProperty<Long> {
+
+            @Override
+            default Long apply(Object o) {
+                return o instanceof Number ? ((Number) o).longValue() : null;
+            }
+        }
+
+        @FunctionalInterface
+        interface IntegerProperty extends AdditionalProperty<Integer> {
+
+            @Override
+            default Integer apply(Object o) {
+                return o instanceof Number ? ((Number) o).intValue() : null;
+            }
+        }
+
+        /**
+         * Extracted from the query <code>fetch</code> parameter or from the server default.
+         * <pre>
+         * Not part of the standard spec, but returned by openEHR since version <code>1.0.0</code> for debugging purposes.
+         * </pre>
+         */
+        LongProperty fetch = () -> "fetch";
```

Comment:
If fetch is long than so would resultSize

Reply:
you are right, integer seems to be sufficient.